### DISCUSSION TITLE
profile condition problem (aspect ratio)

### DISCUSSION THREAD

**Discussion by geextahslex** [mpv-player/mpv#14362] [Q&A] [answered]
Hi I have a problem with setting up my conditions correct. Depending on the aspect ratio I want a different video-crop. If it's between 1.78 and 2.2 I want it to be `video-crop=100%x95%` but if it's above 2.2 I want it to get back to normal `video-crop=100%x100%`
Now the OSC3 profile doesn't work, if a video has an aspect ratio above 2.2 it's still at `video-crop=100%x95%`
I guess I need to limit OSC2 to 1.78 to 2.2 but I don't know how to do that.

Does someone know how to tackle this?
Thank you :)
```
[OSC2]
profile-cond=width / height >=1.78 and get("video-params/primaries") == "bt.2020"
profile-restore=copy
video-crop=100%x95%

[OSC3]
profile-cond=width / height >=2.2 and get("video-params/primaries") == "bt.2020"
profile-restore=copy
video-crop=100%x100%
```

**Comment by Akemi** [comment, marked as answer]
this is just a guess, but probably because >=2.2 is also >=1.78 and both profiles apply. depending on the priority only the former or the latter does anything (they overwrite each other). you probably should test a range eg `width / height >=1.78 and width / height <2.2` for your OSC2 profile, to exclude the case from OSC3.

**Reply by geextahslex** [reply]
Thank you, exactly what I was looking for :)

**Comment by guidocella** [comment]
Swap the profiles, the first one has higher priority.

**Reply by geextahslex** [reply]
Thank you :)
